lib/optimizely/optimizely_user_context.rb

```diff
@@ -50,7 +58,7 @@ def set_attribute(attribute_key, attribute_value)
     # @return [OptimizelyDecision] A decision result
 
     def decide(key, options = nil)
-      @optimizely_client&.decide(self, key, options)
+      @optimizely_client&.decide(clone, key, options)
```

Comment:
why passing `clone`, we already cloning `user_attributes`

Reply:
These are two separate things.
1. Clone is used to pass a full clone of userContext so that subsequect updates to attributes has no impact on in flight decisions and the final userContext object which is referred to in decision object has the same attribute information that was used for the decision.
2. user_attributes cloning is for the scenario where some access user_attributes for some other use from out side. We want to make sure that the copy that they read is not mutated by setAttribute called somewherre later in the code.